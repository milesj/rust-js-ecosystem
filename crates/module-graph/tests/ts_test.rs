mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::*;

mod ts {
    use super::*;

    #[test]
    fn export_star() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "ts/export-star.ts"));
    }

    #[test]
    fn export_named() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-named.ts"
        ));
    }

    #[test]
    fn export_default_function() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-func.ts"
        ));
        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-anon-func.ts"
        ));
    }

    #[test]
    fn export_default_class() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-class.ts"
        ));
        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-anon-class.ts"
        ));
    }

    #[test]
    fn export_default_reference() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-ref.ts"
        ));
    }

    #[test]
    fn export_default_interface() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-interface.ts"
        ));
    }

    // This doesn't seem to generate exports,
    // even though oxc is a valid enum type!
    #[test]
    fn export_default_enum() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-enum.ts"
        ));
    }
}
