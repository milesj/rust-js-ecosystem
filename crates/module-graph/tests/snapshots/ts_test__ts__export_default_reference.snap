---
source: crates/module-graph/tests/ts_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"ts/export-def-ref.ts\")"
---
{
    1: Module {
        exports: [
            Export {
                kind: Modern,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 14,
                        end: 33,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: Default,
                        symbol_id: None,
                        name: "Foo",
                    },
                ],
                type_only: false,
            },
        ],
        fragment: None,
        id: 1,
        imports: [],
        package_name: None,
        path: "/root/ts/export-def-ref.ts",
        query: None,
        source: JavaScriptModule {
            package_type: Unknown,
            source: "class Foo {}\n\nexport default Foo;\n",
            source_type: SourceType {
                language: TypeScript {
                    is_definition_file: false,
                },
                module_kind: Module,
                variant: Standard,
                always_strict: false,
            },
            stats: JavaScriptStats {
                dynamic_import_count: 0,
                export_statements: 1,
                exports_default: true,
                import_statements: 0,
                other_statements: 1,
                require_count: 0,
            },
        },
    },
}
